bigframes/core/blocks.py

```diff
@@ -117,19 +117,6 @@ def __init__(
                     f"'index_columns' (size {len(index_columns)}) and 'index_labels' (size {len(index_labels)}) must have equal length"
                 )
 
-        # If no index columns are set, create one.
-        #
-        # Note: get_index_cols_and_uniqueness in
-        # bigframes/session/_io/bigquery/read_gbq_table.py depends on this
-        # being as sequential integer index column. If this default behavior
-        # ever changes, please also update get_index_cols_and_uniqueness so
-        # that users who explicitly request a sequential integer index can
-        # still get one.
-        if len(index_columns) == 0:
```

Comment:
Would this be a good opportunity to raise some subclass of PreviewWarning? I'm not 100% confident that we've covered all the cases in tests well enough to call this feature "GA".

Reply:
Added preview warning